### ISSUE TITLE
Named export (styled) from "linaria/react" and rollup produce invalid output

### ISSUE THREAD

**Issue by wereHamster** [callstack/linaria#543]
## Description

Given the following code: 

```
import { styled } from "linaria/react";
console.log(styled.h1`color: red`);
```

When you compile it with rollup it will produce output that crashes at runtime.

The crash happens because the import from "@emotion/is-prop-valid" is not processed correctly.

The generated code is something along these lines:

```
/*** NOTE: this is the code from "@emotion/is-prop-valid" ***/
var index = memoize(function (prop) {
  return reactPropsRegex.test(prop) || prop.charCodeAt(0) === 111
  /* o */
  && prop.charCodeAt(1) === 110
  /* n */
  && prop.charCodeAt(2) < 91;
}
/* Z+1 */
);

/*** SNIP ***/

var validAttr = index.default;

/*** SNIP ***/

if (_key === 'as' || validAttr(_key)) {
```

Notice how the code creates the memoized function (`index`), but then uses the `.default` prop on it, which makes `validAttr` undefined. Later the code crashes because it tries to call that undefined function.

I also noticed that linaria doesn't provide ESM entry points, only CommonJS 😕

## Reproducible Demo

Use the above code with this rollup config:

```
import commonjs from "@rollup/plugin-commonjs";
import resolve from "@rollup/plugin-node-resolve";
import babel from "rollup-plugin-babel";

export default {
    input: "in.js",
    output: [
      {
        file: "'out.js",
        format: "esm"
      }
    ],
    plugins: [
      resolve(),
      commonjs({
        namedExports: {
          "linaria/react": ["styled"]
        }
      }),
      babel({
        presets: [["linaria/babel"]]
      })
    ]
  }
```

**Comment by wereHamster** [comment]
Code to reproduce: https://github.com/wereHamster/linaria-react-rollup-bug

**Comment by lucat1** [comment]
Same here, running the fresh rollup `v2.0.0` if that matters

**Comment by TMaszko** [comment]
Thanks, that you spotted this bug and found exact place where it occured. It seems to be fixed in 1.4 and 2.0 release which is currently still in beta but will be released soon :)

**Comment by wereHamster** [comment]
I can tentatively confirm that it's working in 2.0.0-alpha.5. However I can't definitely upgrade to 2.0.0 until #599 is fixed (PR here: #604).

**Comment by TMaszko** [comment]
@wereHamster see explanation here :) https://github.com/rollup/rollup/issues/3424#issuecomment-630813755
